vars/pipelineFairgroundTopUpBots.groovy

```diff
@@ -0,0 +1,111 @@
+/* groovylint-disable DuplicateStringLiteral */
+/* groovylint-disable DuplicateNumberLiteral */
+/* groovylint-disable MethodSize */
+/* groovylint-disable NestedBlockDepth */
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+
+void call() {
+
+    properties([
+        copyArtifactPermission('*'),
+        disableConcurrentBuilds(),
+        pipelineTriggers([cron('H */4 * * *')]),
+        parameters([
+            string(
+                name: 'DEVOPS_INFRA_BRANCH', defaultValue: pipelineDefaults.fair.devopsInfraBranch,
+                description: 'Git branch, tag or hash of the vegaprotocol/devops-infra repository'),
+        ])
+    ])
+
+    echo "params=${params}"
+
+    node {
+        // NOTE: environment variables PSSH_USER and PSSH_KEYFILE are used by veganet.sh script
+        /* groovylint-disable-next-line NoDef, VariableTypeRequired */
+        def sshFairgroundCredentials = sshUserPrivateKey(  credentialsId: 'ssh-vega-network',
+                                                     keyFileVariable: 'PSSH_KEYFILE',
+                                                    usernameVariable: 'PSSH_USER')
+        Map dockerCredentials = [credentialsId: 'github-vega-ci-bot-artifacts',
+                                           url: 'https://ghcr.io']
+        skipDefaultCheckout()
+        cleanWs()
+
+        timestamps {
+            try {
+                timeout(time: 15, unit: 'MINUTES') {
+                    stage('CI config') {
+                        // Printout all configuration variables
+                        sh 'printenv'
```

Comment:
nonblocking: We can think in future to replace print env with something else to avoid printing env variable values exported in the system which may be sensitive.